app/services/zip_delivery_service.rb

```diff
@@ -15,8 +15,22 @@ def initialize(s3_part:, dvz_part:, metadata:)
     @metadata = stringify_values(metadata)
   end
 
-  def deliver
-    return if s3_part.exists?
+  def deliver # rubocop:disable Metrics/AbcSize
+    if s3_part.exists?
+      msg = 'WARNING: S3 location already has content. ' \
+            'Perhaps a failed replication was pruned from the database, but was not removed from the cloud. ' \
+            "If so, prune this content from the database again (there's a rake task), ask ops to delete the bad replicated content, and try again."
+      hb_context = {
+        druid: dvz_part.druid.druid, # dvz_part.dvz.druid is a DruidTools::Druid object
```

Comment:
Does this given enough context to know what the S3 location and part is that already has content? I know you're trying to provide additional information (thank you!), but I'm still not following what the S3 location is and what the part is.

Happy to discuss more because you can't be expected to guess what I don't understand.

Reply:
dvz_part is the DruidVersionZipPart so we can get the druid and the version in question (what would need to be redone) and s3_part is an aws-sdk  gem object and the bucket_name should give us the endpoint.  That should be enough (and is what John put in the ticket.)